### DISCUSSION TITLE
Is Tuya Local truly local control?

### DISCUSSION THREAD

**Discussion by Daverover66** [make-all/tuya-local#720] [Q&A] [answered]
I have quite a few wifi Tuya devices, mostly smart plugs, that are running in Tuya Local.  I presumed this form of local control would negate the need for cloud control through the Smart Life app. However, today,  one of my smart plugs stopped working. It would turn off immediately after being turned on manually.  I wondered if this was a symptom of the glitch in the official Tuya integration in Home Assistant so I thought I'd try deleting the device from the Smart Life app to prevent it interfering with local control. Imagine my surprise when the local device also disappeared from Home Assistant.  I added the device back into the Smart Life app, but as yet I've not been able to regain local control of the device. The local key seems to have changed and the Tuya Local config won't accept either the old local key or the new one.

So, if the local device becomes unavailable in Home Assistant after removing it from the Smart Life app, how can it truly be local control?

**Comment by make-all** [comment, marked as answer]
Tuya devices go back into pairing mode if they are aware that they have been removed from the app they were paired with.  There may be ways to block them from finding this out, but some devices also re-enter pairing mode if they cannot connect to the cloud for an extended period too.

Although this integration is purely local, it cannot influence the Tuya device behaviour, which is hybrid cloud/local.
